code/datums/gamemodes/crash.dm

```diff
@@ -0,0 +1,289 @@
+/datum/game_mode/crash
+	name = "Crash"
+	config_tag = "Crash"
+	required_players = 0
+	flags_round_type = MODE_INFESTATION|MODE_FOG_ACTIVATED
+	flags_landmarks = MODE_LANDMARK_SPAWN_XENO_TUNNELS|MODE_LANDMARK_SPAWN_MAP_ITEM
+
+	round_end_states = list(MODE_CRASH_X_MAJOR, MODE_CRASH_M_MAJOR, MODE_CRASH_X_MINOR, MODE_CRASH_M_MINOR, MODE_CRASH_DRAW_DEATH)
+
+	// Round start conditions
+	var/xeno_required_num = 1
+	var/xeno_starting_num = 0
+	var/list/xenomorphs
+	
+	// Round end conditions
+	var/marines_evac = FALSE
+	var/planet_nuked = FALSE
+
+	var/shuttle_id = "tgs_canterbury"
+	var/obj/docking_port/mobile/crashmode/shuttle
+
+
+/datum/game_mode/crash/New()
+	. = ..()
+
+	xenomorphs = list()
+
+
+/datum/game_mode/crash/can_start()
+	. = ..()
+	// Check if enough players have signed up for xeno & queen roles.
+	initialize_xeno_leader()
+	initialize_xenomorphs()
+	if(!length(xenomorphs)) // we need at least 1
+		return FALSE
+
+	return TRUE
+
+
+/datum/game_mode/crash/pre_setup()
+	. = ..()
+	// Spawn the ship
+	if(!SSmapping.shuttle_templates[shuttle_id])
+		message_admins("Gamemode: Shuttle [shuttle_id] wasn't found and can't be loaded")
+		CRASH("Shuttle [shuttle_id] wasn't found and can't be loaded")
+		return FALSE
+
+	var/datum/map_template/shuttle/S = SSmapping.shuttle_templates[shuttle_id]
+	var/obj/docking_port/stationary/L = SSshuttle.getDock("crashmodeloading")
+	shuttle = SSshuttle.action_load(S, L)
+
+	// Reset all spawnpoints after spawning the ship
+	GLOB.marine_spawns_by_job = shuttle.marine_spawns_by_job
+	GLOB.jobspawn_overrides = list()
+	GLOB.latejoin = shuttle.spawnpoints
+	GLOB.latejoin_cryo = shuttle.spawnpoints
+	GLOB.latejoin_gateway = shuttle.spawnpoints
+
+	// Create xenos
+	var/number_of_xenos = length(xenomorphs)
+	var/datum/hive_status/normal/HN = GLOB.hive_datums[XENO_HIVE_NORMAL]
+	for(var/i in xenomorphs)
+		var/datum/mind/M = i
+		if(M.assigned_role == ROLE_XENO_QUEEN)
+			transform_ruler(M, number_of_xenos > HN.xenos_per_queen)
+		else
+			transform_xeno(M)
+
+
+/datum/game_mode/crash/setup()
```

Comment:
This is the exact same as the parent's setup

Reply:
It wasn't originally but yeah i can remove it now.